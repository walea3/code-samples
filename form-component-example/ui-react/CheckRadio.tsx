import cx from 'classnames';
import React, { ReactNode } from 'react';

import { getDataRef } from '@nel-ui/js-helpers';
import { CheckRadioIcon } from './CheckRadioIcon';
import FormStyles from './FormStyles';

// TODO: Remove inline here when support is added
export type CheckRadioSingular = Omit<CheckRadioProps, 'inline' | 'variant'>;

export interface CheckRadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
  id: string;
  label: ReactNode;
  variant: 'checkbox' | 'radio';
  alert?: 'warning' | 'error';
  inline?: boolean;
  disabled?: boolean;
  'data-ref'?: string;
}

export function CheckRadio({
  'data-ref': dataRef,
  variant,
  label,
  alert,
  inline,
  className,
  ...props
}: CheckRadioProps) {
  return (
    <FormStyles>
      {({ classes }) => (
        <div
          className={cx(classes.checkRadioContainer, {
            [classes.inlineCheckRadio]: inline,
          })}
          data-ref={getDataRef(dataRef, `${variant}Container`)}
        >
          <input
            data-ref={getDataRef(dataRef, `${variant}Input`)}
            type={variant}
            className={classes.checkRadioInput}
            aria-invalid={!!alert}
            {...props}
          />

          <label
            data-ref={getDataRef(dataRef, variant)}
            htmlFor={props.id}
            className={cx(className, classes.checkRadio, classes[variant], {
              [classes[`${alert}CheckRadio`]]: alert,
            })}
          >
            <span
              data-ref={getDataRef(dataRef, `${variant}Control`)}
              className={classes.checkRadioControl}
            >
              <CheckRadioIcon variant={variant} />
            </span>
            {label}
          </label>
        </div>
      )}
    </FormStyles>
  );
}
