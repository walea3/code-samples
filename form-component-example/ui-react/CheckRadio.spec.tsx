import { renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import React from 'react';

import { Checkbox, CheckRadioVariantProps, Radio } from '.';
import { CheckRadio, CheckRadioProps } from './CheckRadio';

describe('@renders', () => {
  const requiredCheckboxProps = {
    id: 'checkbox-test',
    label: 'checkbox-test',
    variant: 'checkbox',
  };
  const requiredRadioProps = {
    id: 'radio-test',
    label: 'radio-test',
    variant: 'radio',
  };

  const titleNode: CheckRadioVariantProps['label'] = (
    <span>
      a <strong>custom</strong> title
    </span>
  );

  const renderComponent = (props: CheckRadioProps) =>
    renderWithNEL(<CheckRadio {...props} />);

  test.each`
    name                               | props
    ${'as checkbox'}                   | ${requiredCheckboxProps}
    ${'as checkbox with custom title'} | ${{ ...requiredCheckboxProps, label: titleNode }}
    ${'with warning as checkbox'}      | ${{ ...requiredCheckboxProps, alert: 'warning' }}
    ${'with error as checkbox'}        | ${{ ...requiredCheckboxProps, alert: 'error' }}
    ${'with data-ref as checkbox'}     | ${{ ...requiredCheckboxProps, 'data-ref': 'uid' }}
    ${'as radio'}                      | ${requiredRadioProps}
    ${'as radio with custom title'}    | ${{ ...requiredRadioProps, label: titleNode }}
    ${'with warning as radio'}         | ${{ ...requiredRadioProps, alert: 'warning' }}
    ${'with error as radio'}           | ${{ ...requiredRadioProps, alert: 'error' }}
    ${'with data-ref as radio'}        | ${{ ...requiredRadioProps, 'data-ref': 'uid' }}
  `('$name', ({ props }) => {
    expect(renderComponent(props)).toMatchSnapshot();
  });

  describe('variant shorthand', () => {
    const requiredProps = {
      id: 'variantTest',
      label: 'variantTest',
    };

    test.each`
      name              | Component
      ${'<Checkbox />'} | ${Checkbox}
      ${'<Radio />'}    | ${Radio}
    `('@renders $name', ({ Component }) => {
      expect(renderWithNEL(<Component {...requiredProps} />)).toMatchSnapshot();
    });
  });
});
