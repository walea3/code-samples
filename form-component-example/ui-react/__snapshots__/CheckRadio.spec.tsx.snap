// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@renders as checkbox 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="checkboxContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="checkboxInput"
    id="checkbox-test"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox"
    data-ref="checkbox"
    for="checkbox-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    checkbox-test
  </label>
</div>
`;

exports[`@renders as checkbox with custom title 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="checkboxContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="checkboxInput"
    id="checkbox-test"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox"
    data-ref="checkbox"
    for="checkbox-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    <span>
      a 
      <strong>
        custom
      </strong>
       title
    </span>
  </label>
</div>
`;

exports[`@renders as radio 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="radioContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="radioInput"
    id="radio-test"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio"
    data-ref="radio"
    for="radio-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    radio-test
  </label>
</div>
`;

exports[`@renders as radio with custom title 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="radioContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="radioInput"
    id="radio-test"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio"
    data-ref="radio"
    for="radio-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    <span>
      a 
      <strong>
        custom
      </strong>
       title
    </span>
  </label>
</div>
`;

exports[`@renders variant shorthand @renders <Checkbox /> 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="checkboxContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="checkboxInput"
    id="variantTest"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox"
    data-ref="checkbox"
    for="variantTest"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    variantTest
  </label>
</div>
`;

exports[`@renders variant shorthand @renders <Radio /> 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="radioContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="radioInput"
    id="variantTest"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio"
    data-ref="radio"
    for="variantTest"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    variantTest
  </label>
</div>
`;

exports[`@renders with data-ref as checkbox 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="uid.checkboxContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="uid.checkboxInput"
    id="checkbox-test"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox"
    data-ref="uid.checkbox"
    for="checkbox-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="uid.checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    checkbox-test
  </label>
</div>
`;

exports[`@renders with data-ref as radio 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="uid.radioContainer"
>
  <input
    aria-invalid="false"
    class="nel-checkRadioInput"
    data-ref="uid.radioInput"
    id="radio-test"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio"
    data-ref="uid.radio"
    for="radio-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="uid.radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    radio-test
  </label>
</div>
`;

exports[`@renders with error as checkbox 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="checkboxContainer"
>
  <input
    aria-invalid="true"
    class="nel-checkRadioInput"
    data-ref="checkboxInput"
    id="checkbox-test"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox nel-errorCheckRadio"
    data-ref="checkbox"
    for="checkbox-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    checkbox-test
  </label>
</div>
`;

exports[`@renders with error as radio 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="radioContainer"
>
  <input
    aria-invalid="true"
    class="nel-checkRadioInput"
    data-ref="radioInput"
    id="radio-test"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio nel-errorCheckRadio"
    data-ref="radio"
    for="radio-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    radio-test
  </label>
</div>
`;

exports[`@renders with warning as checkbox 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="checkboxContainer"
>
  <input
    aria-invalid="true"
    class="nel-checkRadioInput"
    data-ref="checkboxInput"
    id="checkbox-test"
    type="checkbox"
  />
  <label
    class="nel-checkRadio nel-checkbox nel-warningCheckRadio"
    data-ref="checkbox"
    for="checkbox-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="checkboxControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
        />
      </svg>
    </span>
    checkbox-test
  </label>
</div>
`;

exports[`@renders with warning as radio 1`] = `
<div
  class="nel-checkRadioContainer"
  data-ref="radioContainer"
>
  <input
    aria-invalid="true"
    class="nel-checkRadioInput"
    data-ref="radioInput"
    id="radio-test"
    type="radio"
  />
  <label
    class="nel-checkRadio nel-radio nel-warningCheckRadio"
    data-ref="radio"
    for="radio-test"
  >
    <span
      class="nel-checkRadioControl"
      data-ref="radioControl"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
        />
      </svg>
    </span>
    radio-test
  </label>
</div>
`;
